core/src/main/java/org/elasticsearch/common/settings/SecureString.java

```diff
@@ -51,21 +51,19 @@ public SecureString(char[] chars) {
      * Constructs a new SecureString from a StreamInput.
      */
     public SecureString(StreamInput in) throws IOException {
-        ByteBuffer bytes = null;
-        CharBuffer charBuffer = null;
+        BytesRef ref = null;
+        char[] chars = null;
         try {
-            final int length = in.readVInt();
-            final byte[] byteArray = new byte[length];
-            in.readBytes(byteArray, 0, length);
-            bytes = ByteBuffer.wrap(byteArray);
-            charBuffer = StandardCharsets.UTF_8.decode(bytes);
-            this.chars = Arrays.copyOfRange(charBuffer.array(), charBuffer.arrayOffset(), charBuffer.limit());
+            ref = in.readBytesRef();
```

Comment:
you can move this out of the try block and make both the ref and the chars final. Since if the `in. readBytesRef()` fails you have nothing to null out and the `new char[ref.length]` won't fail and if so it's fatal I guess. then the null checks can go as well below.